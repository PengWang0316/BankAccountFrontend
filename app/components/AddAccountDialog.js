import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import I18n from '@kevinwang0316/i18n';

import { addAccount as addAccountAction } from '../actions/AccountActions';

export const AddAccountDialog = ({ open, handleClose, addAccount }) => {
  const [name, setName] = useState('');
  const [balance, setBalance] = useState(0);

  const handleAddAccount = () => {
    addAccount({ name, balance });
    setName('');
    setBalance(0);
    handleClose();
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">{I18n.get('addAccountDialogTitle')}</DialogTitle>
      <DialogContent>
        <TextField
          autoFocus
          margin="dense"
          id="name"
          label="Name"
          type="text"
          fullWidth
          value={name}
          onChange={({ target }) => setName(target.value)}
        />
        <TextField
          margin="dense"
          id="balance"
          label="Balance"
          type="number"
          fullWidth
          value={balance}
          onChange={({ target }) => setBalance(target.value)}
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('cancel')}
        </Button>
        <Button onClick={handleAddAccount} color="primary">
          {I18n.get('add')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
AddAccountDialog.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func.isRequired,
  addAccount: PropTypes.func.isRequired,
};
AddAccountDialog.defaultProps = { open: false };

/* istanbul ignore next */
const mapDispatchToProps = { addAccount: addAccountAction };
export default connect(null, mapDispatchToProps)(AddAccountDialog);
