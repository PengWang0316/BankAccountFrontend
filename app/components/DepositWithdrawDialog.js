import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import I18n from '@kevinwang0316/i18n';

import {
  deposit as depositAction, withdraw as withdrawAction,
} from '../actions/AccountActions';

export const DepositWithdrawDialog = ({
  open, handleClose, isDeposit, accountId, deposit, withdraw,
}) => {
  const [amount, setAmount] = useState(0);

  const handleDeposit = () => deposit(amount, accountId);

  const handleWithdraw = () => withdraw(amount, accountId);

  const handleConfirmClick = () => {
    if (isDeposit) handleDeposit();
    else handleWithdraw();
    setAmount(0);
    handleClose();
  };

  return (
    <Dialog
      open={open}
      onClose={handleClose}
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">{I18n.get(isDeposit ? 'depositDialogTitle' : 'withdrawDialogTitle')}</DialogTitle>
      <DialogContent>
        <DialogContentText>
          {`${isDeposit ? I18n.get('depositDialogText') : I18n.get('withdrawDialogText')} ${accountId}`}
        </DialogContentText>
        <TextField
          autoFocus
          margin="dense"
          id="amount"
          label={I18n.get('thAmount')}
          type="number"
          fullWidth
          value={amount}
          onChange={({ target }) => setAmount(target.value)}
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          {I18n.get('cancel')}
        </Button>
        <Button onClick={handleConfirmClick} color="primary">
          {I18n.get(isDeposit ? 'deposit' : 'withdraw')}
        </Button>
      </DialogActions>
    </Dialog>
  );
};
DepositWithdrawDialog.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func.isRequired,
  deposit: PropTypes.func.isRequired,
  withdraw: PropTypes.func.isRequired,
  isDeposit: PropTypes.bool.isRequired,
  accountId: PropTypes.string.isRequired,
};
DepositWithdrawDialog.defaultProps = { open: false };

/* istanbul ignore next */
const mapDispatchToProps = {
  withdraw: withdrawAction,
  deposit: depositAction,
};
export default connect(null, mapDispatchToProps)(DepositWithdrawDialog);
