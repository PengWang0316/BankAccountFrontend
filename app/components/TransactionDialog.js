import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';

import LoadingAnimation from './SharedComponents/LoadingAnimation';

/* istanbul ignore next */
const styles = theme => ({
  paper: {
    width: '90%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    margin: 'auto',
    // minHeight: 400,
  },
  table: {
    minWidth: 800,
    // minHeight: 400,
  },
});

export const TransactionDialog = ({
  open, handleClose, transactions, classes, accountId,
}) => (
  <Dialog
    open={open}
    onClose={handleClose}
    aria-labelledby="form-dialog-title"
    maxWidth="md"
  >
    <DialogTitle id="form-dialog-title">{I18n.get('transicationDialogTitle')}</DialogTitle>
    <DialogContent>
      <Paper className={classes.paper}>
        <Table className={classes.table}>
          <TableHead>
            <TableRow>
              <TableCell padding="dense">{I18n.get('thAccountId')}</TableCell>
              <TableCell padding="dense">{I18n.get('thType')}</TableCell>
              <TableCell padding="dense">{I18n.get('thAmount')}</TableCell>
              <TableCell align="justify" padding="dense">{I18n.get('thDate')}</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {!transactions && (
              <TableRow><LoadingAnimation /></TableRow>
            )}
            {transactions && transactions.map(({
              key, amount, type, date,
            }) => (
              <TableRow key={key}>
                <TableCell padding="dense">{accountId}</TableCell>
                <TableCell padding="dense">{type}</TableCell>
                <TableCell padding="dense">${amount}</TableCell>
                <TableCell padding="dense">{date}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </Paper>
    </DialogContent>
    <DialogActions>
      <Button onClick={handleClose} color="primary">
        {I18n.get('close')}
      </Button>
    </DialogActions>
  </Dialog>
);
TransactionDialog.propTypes = {
  open: PropTypes.bool,
  handleClose: PropTypes.func.isRequired,
  accountId: PropTypes.string.isRequired,
  transactions: PropTypes.arrayOf(PropTypes.object),
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};
TransactionDialog.defaultProps = { open: false, transactions: null };

export default withStyles(styles)(TransactionDialog);
