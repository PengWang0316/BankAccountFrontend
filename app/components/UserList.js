/* eslint-disable react/jsx-one-expression-per-line */
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import I18n from '@kevinwang0316/i18n';

import { fetchAllAccount as fetchAllAccountsAction } from '../actions/AccountActions';

const styles = theme => ({
  root: {
    width: '70%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    margin: 'auto',
    // minHeight: 400,
  },
  table: {
    minWidth: 700,
    // minHeight: 400,
  },
});

const UserList = ({ classes, accounts, fetchAllAccount }) => {
  useEffect(() => {
    if (!accounts) fetchAllAccount();
  });
  return (
    <Paper className={classes.root}>
      <Table className={classes.table}>
        <TableHead>
          <TableRow>
            <TableCell>{I18n.get('thAccountId')}</TableCell>
            <TableCell>{I18n.get('thName')}</TableCell>
            <TableCell>{I18n.get('thBalance')}</TableCell>
            <TableCell align="right">{I18n.get('thActions')}</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {accounts && Object.keys(accounts).map(key => (
            <TableRow key={key} id={key}>
              <TableCell>{accounts[key].accountId}</TableCell>
              <TableCell>{accounts[key].name}</TableCell>
              <TableCell>${accounts[key].balance}</TableCell>
              <TableCell align="right">
                <Button size="small" color="primary">{I18n.get('deposit')}</Button><Button size="small" color="secondary">{I18n.get('withdraw')}</Button>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </Paper>
  );
};
UserList.propTypes = {
  accounts: PropTypes.object,
  fetchAllAccount: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
};
UserList.defaultProps = { accounts: null };

const mapStateToProps = ({ accounts }) => ({ accounts });
const mapDispatchToProps = { fetchAllAccount: fetchAllAccountsAction };
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(UserList));
