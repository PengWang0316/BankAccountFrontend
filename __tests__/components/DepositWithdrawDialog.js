import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { DepositWithdrawDialog } from '../../app/components/DepositWithdrawDialog';

jest.mock('@material-ui/core/Button', () => 'Button');
jest.mock('@material-ui/core/TextField', () => 'TextField');
jest.mock('@material-ui/core/Dialog', () => 'Dialog');
jest.mock('@material-ui/core/DialogActions', () => 'DialogActions');
jest.mock('@material-ui/core/DialogContent', () => 'DialogContent');
jest.mock('@material-ui/core/DialogTitle', () => 'DialogTitle');
jest.mock('@kevinwang0316/i18n', () => ({ get: key => key }));

describe('DepositWithdrawDialog', () => {
  const defaultProps = {
    open: false,
    handleClose: jest.fn(),
    deposit: jest.fn(),
    withdraw: jest.fn(),
    isDeposit: true,
    accountId: 'accountId',
  };
  const getShallowComponent = (props = defaultProps) => shallow(<DepositWithdrawDialog {...props} />);

  beforeEach(() => jest.clearAllMocks());

  test('Initial state', () => {
    const component = getShallowComponent();

    const textFileds = component.find('TextField');
    expect(textFileds.prop('value')).toBe(0);
  });

  test('handleClose', () => {
    const component = getShallowComponent();
    component.find('Button').at(0).simulate('click');

    expect(defaultProps.handleClose).toHaveBeenCalledTimes(1);
  });

  test('handleClickDeposit', () => {
    const component = getShallowComponent();
    const textFileds = component.find('TextField');
    textFileds.simulate('change', { target: { value: 100 } });
    component.find('Button').at(1).simulate('click');

    expect(defaultProps.deposit).toHaveBeenCalledTimes(1);
    expect(defaultProps.deposit).toHaveBeenLastCalledWith(100, 'accountId');
    expect(defaultProps.handleClose).toHaveBeenCalledTimes(1);
  });

  test('handleClickWithdraw', () => {
    const component = getShallowComponent({ ...defaultProps, isDeposit: false });
    const textFileds = component.find('TextField');
    textFileds.simulate('change', { target: { value: 100 } });
    component.find('Button').at(1).simulate('click');

    expect(defaultProps.withdraw).toHaveBeenCalledTimes(1);
    expect(defaultProps.withdraw).toHaveBeenLastCalledWith(100, 'accountId');
    expect(defaultProps.handleClose).toHaveBeenCalledTimes(1);
  });

  test('DepositWithdrawDialog with deposit snapshot', () => expect(renderer.create(<DepositWithdrawDialog {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('DepositWithdrawDialog with withdraw snapshot', () => expect(renderer.create(<DepositWithdrawDialog {...{ ...defaultProps, isDeposit: false }} />).toJSON()).toMatchSnapshot());
});
