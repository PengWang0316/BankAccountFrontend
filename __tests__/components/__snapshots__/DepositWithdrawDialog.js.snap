// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositWithdrawDialog DepositWithdrawDialog with deposit snapshot 1`] = `
<Dialog
  aria-labelledby="form-dialog-title"
  onClose={[MockFunction]}
  open={false}
>
  <DialogTitle
    id="form-dialog-title"
  >
    depositDialogTitle
  </DialogTitle>
  <DialogContent>
    <p
      className="MuiTypography-root-2 MuiDialogContentText-root-1 MuiTypography-subheading-9 MuiTypography-colorTextSecondary-35"
    >
      depositDialogText accountId
    </p>
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="amount"
      label="thAmount"
      margin="dense"
      onChange={[Function]}
      type="number"
      value={0}
    />
  </DialogContent>
  <DialogActions>
    <Button
      color="primary"
      onClick={[MockFunction]}
    >
      cancel
    </Button>
    <Button
      color="primary"
      onClick={[Function]}
    >
      deposit
    </Button>
  </DialogActions>
</Dialog>
`;

exports[`DepositWithdrawDialog DepositWithdrawDialog with withdraw snapshot 1`] = `
<Dialog
  aria-labelledby="form-dialog-title"
  onClose={[MockFunction]}
  open={false}
>
  <DialogTitle
    id="form-dialog-title"
  >
    withdrawDialogTitle
  </DialogTitle>
  <DialogContent>
    <p
      className="MuiTypography-root-2 MuiDialogContentText-root-1 MuiTypography-subheading-9 MuiTypography-colorTextSecondary-35"
    >
      withdrawDialogText accountId
    </p>
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="amount"
      label="thAmount"
      margin="dense"
      onChange={[Function]}
      type="number"
      value={0}
    />
  </DialogContent>
  <DialogActions>
    <Button
      color="primary"
      onClick={[MockFunction]}
    >
      cancel
    </Button>
    <Button
      color="primary"
      onClick={[Function]}
    >
      withdraw
    </Button>
  </DialogActions>
</Dialog>
`;
